persistence/src/main/resources/sql/db_functions.sql

```diff
@@ -0,0 +1,110 @@
+DROP FUNCTION IF EXISTS fn_getCertificatesWithTags(
+    tag_name VARCHAR(50),
+    context VARCHAR(255));
+---------------------------------------------------
+create function fn_getCertificatesWithTags(
+    tag_name VARCHAR(50),
+    context VARCHAR(255)
+)
+    returns table
+            (
+                id               bigint,
+                name             varchar(255),
+                description      varchar(255),
+                price            double precision,
+                duration         int,
+                create_date      timestamptz,
+                last_update_date timestamptz,
+                isDeleted        int,
+                tagName          varchar(255)
+            )
+as
+$$
+begin
+
+    case
+
+        when ((tag_name IS NOT NULL) AND (context IS NOT NULL)) then
```

Comment:
You could get rid of these cases and conditions using check for tagName and context inside specification and passing "%" inside the function in case of empty value. Try SIMILAR TO '%' || context || '%' and same for tagName.